### PR TITLE
Export 500 gene variants resulting from SNVs and Indels queries

### PR DESCRIPTION
**Pull request by northwestwitch**
Partial fix for #3946

New button to export variants from `Search SNVs and Indels` Page:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
I wanted to test this one on cg-vm1 but We are blocked by #4005

**Comment by dnil** [comment]
> I wanted to test this one on cg-vm1 but We are blocked by #4005

I'll be there in an h or so! 😊 Perhaps you could request a review from @Vince-janv for #4005 if you need it sooner..

**Comment by dnil** [comment]
> > I wanted to test this one on cg-vm1 but We are blocked by #4005
> 
> I'll be there in an h or so! 😊 Perhaps you could request a review from @Vince-janv for #4005 if you need it sooner..

I "reviewed" - its for stage anyway, works and works for the other services.

**Comment by northwestwitch** [comment]
Looks like it's working as it should now. Ready for review

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 1]
```diff
@@ -36,6 +38,38 @@ def test_gene_variants(app, user_obj, institute_obj):
         assert "POT1" in str(resp.data)
 
 
+def test_gene_variants_export(app, user_obj, institute_obj):
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:41]: 🥇

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 from scout.server.extensions import store
 
+SNPS_INDELS_ENDPOINT = "overview.gene_variants"
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:7]:
```suggestion
OVERVIEW_GENE_VARIANTS_ENDPOINT = "overview.gene_variants"
```

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 3]
```diff
@@ -24,7 +26,7 @@ def test_gene_variants(app, user_obj, institute_obj):
         # WHEN form is submitted by POST request
         resp = client.post(
             url_for(
-                "overview.gene_variants",
+                SNPS_INDELS_ENDPOINT,
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:29]:
```suggestion
                OVERVIEW_GENE_VARIANTS_ENDPOINT,
```

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 4]
```diff
@@ -36,6 +38,38 @@ def test_gene_variants(app, user_obj, institute_obj):
...
+
+    # GIVEN an initialized app
+    with app.test_client() as client:
+        # WITH a logged user
+        client.get(url_for("auto_login"))
+
+        # WHEN form is submitted by POST request
+        resp = client.post(
+            url_for(
+                SNPS_INDELS_ENDPOINT,
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:61]:
```suggestion
                OVERVIEW_GENE_VARIANTS_ENDPOINT,
```

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 5]
```diff
@@ -433,7 +467,7 @@ def test_gene_variants_filter(app, institute_obj, case_obj):
         }
 
         resp = client.post(
-            url_for("overview.gene_variants", institute_id=institute_obj["internal_id"]),
+            url_for(SNPS_INDELS_ENDPOINT, institute_id=institute_obj["internal_id"]),
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:470]:
```suggestion
            url_for(OVERVIEW_GENE_VARIANTS_ENDPOINT, institute_id=institute_obj["internal_id"]),
```

**Review by dnil — APPROVED** [review]
Nice!

File: scout/server/blueprints/institutes/controllers.py [hunk 6]
```diff
@@ -574,6 +577,78 @@ def get_sanger_unevaluated(store, institute_id, user_id):
...
+            str(round(variant.get("cadd_score"), 1)) if variant.get("cadd_score") else "-"
+        )  # CADD score
+        variant_line.append(" | ".join(variant.get("region_annotations", [])))  # Region
+        variant_line.append(" | ".join(variant.get("functional_annotations", [])))  # Function
+        variant_line.append(variant.get("hgvs", "-"))  # HGVS
+
+        export_lines.append(",".join(variant_line))
+
+    headers = Headers()
+    headers.add("Content-Disposition", "attachment", filename="gene_variants.csv")
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/institutes/controllers.py:643]: Institutes can be a lot, if you are an admin or part of many groups, so I would avoid it? But I can add the gene and the datetime perhaps!

**Comment by northwestwitch** [comment]
I gave this PR a final test on the stage server and it's working. Merging!

**Pull request merged**